airbyte-lib/airbyte_lib/factories.py

```diff
@@ -49,20 +49,45 @@ def new_local_cache(
 
 def get_connector(
     name: str,
-    version: str = "latest",
-    config: Optional[Dict[str, Any]] = None,
+    version: str | None = None,
+    pip_url: str | None = None,
+    config: dict[str, Any] | None = None,
     use_local_install: bool = False,
     install_if_missing: bool = False,
 ):
     """
     Get a connector by name and version.
     :param name: connector name
-    :param version: connector version - if not provided, the most recent version will be used
-    :param config: connector config - if not provided, you need to set it later via the set_config method
+    :param version: connector version - if not provided, the most recent version will be used.
+    :param pip_url: connector pip URL - if not provided, the pip url will be inferred from the connector name.
+    :param config: connector config - if not provided, you need to set it later via the set_config method.
     :param use_local_install: whether to use a virtual environment to run the connector. If True, the connector is expected to be available on the path (e.g. installed via pip). If False, the connector will be installed automatically in a virtual environment.
     :param install_if_missing: whether to install the connector if it is not available locally. This parameter is ignored if use_local_install is True.
     """
     metadata = get_connector_metadata(name)
+    if use_local_install:
+        if pip_url:
+            raise ValueError(
+                "Param 'pip_url' is not supported when 'use_local_install' is True"
+            )
+        if version:
+            raise ValueError(
+                "Param 'version' is not supported when 'use_local_install' is True"
+            )
+        executor = PathExecutor(
+            metadata=metadata,
+            target_version=version,
+        )
+
+    else:
+        executor = VenvExecutor(
+            metadata=metadata,
+            target_version=version,
+            install_if_missing=install_if_missing,
+            pip_url=pip_url,
```

Comment:
The added (optional) pip_url can accept any of these:

1. A relative or absolute local path reference.
2. A PyPi package name.
3. A git ref.